patch_recurrence/steps.py

```diff
@@ -76,54 +77,67 @@ def set_patch_buckets(patches, constants):
         map(lambda (i,x): setattr(x, 'bucket', norm_std[k][i]), enumerate(patches[k]))
 
 
-def generate_pairs(patches, constants):
+def generate_pairs(imgs, patches, constants):
     k_nearest = constants.K_NEAREST
-    scaled_imgs = len(patches)      # only those patches with high std deviation
-    # Convert the list of patch norms into numpy arrays
+    patch_size = constants.PATCH_SIZE
+    scaled_imgs = len(patches)
+
     patch_database = []
+    query_database = []
```

Comment:
Perhaps add a comment describing what each of them do? You could combine these lines into `patch_database, candidate_database ... = [], [], ...`